darwin/future/data_objects/team.py

```diff
@@ -51,5 +54,32 @@ class Team(DefaultDarwin):
     # Data Validation
     _slug_validator = validator("slug", allow_reuse=True)(parse_name)
 
+    @staticmethod
+    def from_client(client: Client, team_slug: Optional[str] = None) -> Team:
+        """Returns the team with the given slug"""
+        if not team_slug:
+            team_slug = client.config.default_team
+        return get_team(client, team_slug)
+
 
 TeamList = List[Team]
+
+
+def get_team(client: Client, team_slug: Optional[str] = None) -> Team:
+    """Returns the team with the given slug"""
+    if not team_slug:
+        team_slug = client.config.default_team
+    response = client.get(f"/teams/{team_slug}/")
```

Comment:
We're starting to get a mix of functions that use the monad `exceptions, return` response, and those that raise.  Not something for now, but we should work out what our approach is for using one vs the other.

Reply:
I think for me it comes down to when we get a single object back or a list of objects. If we're only returning a single 'block', ie something that's all related to each other and if one thing goes wrong it's all poisoned, then it makes sense to just return the object + raise if issues, but if we have a function that returns a list of objects and they're all seperate, then it makes sense to package up the good data and return it with the exceptions monad style.